src/terminal_ui.cc

```diff
@@ -555,16 +555,29 @@ void TerminalUI::draw_status(const DisplayLine& status_line,
     if (m_set_title)
     {
         Writer writer{STDOUT_FILENO};
-        constexpr char suffix[] = " - Kakoune\007";
         writer.write("\033]2;");
-        // Fill title escape sequence buffer, removing non ascii characters
-        for (auto& atom : mode_line)
+
+        if (not m_title_line.empty())
         {
-            const auto str = atom.content();
-            for (auto it = str.begin(), end = str.end(); it != end; utf8::to_next(it, end))
-                writer.write((*it >= 0x20 and *it <= 0x7e) ? *it : '?');
+            for (auto ch: m_title_line)
+            {
+                writer.write(ch);
+            }
```

Comment:
writing arbitrary bytes is dangerous, that string could for example contain a `\007` character that would end the sequence. The filtering of all non basic ascii characters is here to avoid this kind of issues and should be preserved.